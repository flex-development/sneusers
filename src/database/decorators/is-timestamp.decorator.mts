/**
 * @file Decorators - IsTimestamp
 * @module sneusers/database/decorators/IsTimestamp
 */

import validator from '#database/decorators/is-timestamp.constaint'
import { ValidateBy, type ValidationOptions } from 'class-validator'

/**
 * Require a value to be a [unix timestamp][timestamp].
 *
 * [timestamp]: https://unixtimestamp.com
 *
 * @decorator
 *
 * @this {void}
 *
 * @param {ValidationOptions | undefined} [options]
 *  Validation options
 * @return {PropertyDecorator}
 *  Property decorator
 */
function IsTimestamp(
  this: void,
  options: ValidationOptions | undefined = {}
): PropertyDecorator {
  return ValidateBy(
    Object.assign({ constraints: [options], validator }, validator.options),
    options
  )
}

export default IsTimestamp
