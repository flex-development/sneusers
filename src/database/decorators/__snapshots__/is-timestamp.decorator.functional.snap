// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`functional:database/decorators/IsTimestamp > should fail if property value is not valid timestamp (0) 1`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "isTimestamp": "value must be a unix timestamp",
    },
    "property": "value",
    "value": -13,
  },
]
`;

exports[`functional:database/decorators/IsTimestamp > should fail if property value is not valid timestamp (1) 1`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "isTimestamp": "value must be a unix timestamp",
    },
    "property": "value",
    "value": null,
  },
]
`;
